src/main/java/de/retest/recheck/RecheckOptions.java

```diff
@@ -1,24 +1,91 @@
 package de.retest.recheck;
 
-import lombok.Builder;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import de.retest.recheck.ignore.CompoundFilter;
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ignore.RecheckIgnoreUtil;
 import lombok.Getter;
-import lombok.experimental.SuperBuilder;
 
-@SuperBuilder
 @Getter
 public class RecheckOptions {
 
-	@Builder.Default
-	private final FileNamerStrategy fileNamerStrategy = new MavenConformFileNamerStrategy();
-
-	@Builder.Default
-	private final String suiteName = null;
+	private final FileNamerStrategy fileNamerStrategy;
+	private final String suiteName;
+	private final boolean reportUploadEnabled;
+	private final Filter filter;
 
-	@Builder.Default
-	private boolean reportUploadEnabled = false;
+	public RecheckOptions( final FileNamerStrategy fileNamerStrategy, final String suiteName,
+			final boolean reportUploadEnabled, final Filter filter ) {
+		this.fileNamerStrategy = fileNamerStrategy;
+		this.suiteName = suiteName;
+		this.reportUploadEnabled = reportUploadEnabled;
+		this.filter = filter;
+	}
 
 	public String getSuiteName() {
```

Comment:
We don't have to trick Lombok anymore, so we should be able to remove this getter and initialize the default within the builder as follows:

```java
private String suiteName = fileNamerStrategy.getTestClassName();
```